ctl/filebuffer.go

```diff
@@ -0,0 +1,116 @@
+package ctl
```

Comment:
Because this one file (`FileBuffer`) seems so stand-alone and general purpose, it seems like it could be in its own package so that another packages could import and use it by calling something like `buffer.NewFileBuffer()` (or whatever its package name might be... `buffer` is probably not right. `files`?). It would also be nice to have its own unit tests.